decimal.go

```diff
@@ -582,6 +581,83 @@ func (d Decimal) Div64(v uint64) (Decimal, error) {
 	return newDecimal(d.neg, bintFromBigInt(dBig), defaultPrec), nil
 }
 
+// QuoRem returns q and r where
+// - q = d / e and  q is an integer
+// - r = d - q * e (r < e and r has the same sign as d)
+//
+// The implementation is similar to C's fmod function.
+// Returns divide by zero error when e is zero
+func (d Decimal) QuoRem(e Decimal) (Decimal, Decimal, error) {
+	if e.coef.IsZero() {
+		return Decimal{}, Decimal{}, ErrDivideByZero
+	}
+
+	q, r, err := tryQuoRemU128(d, e)
+	if err == nil {
+		return q, r, nil
+	}
```

Comment:
Does this method return an error that is not about an overflow?

I'm asking because I find strange arrow is totally in your when not nil. Which means not no matter arrow the process will continue the same way

Reply:
Not sure I get what you mean. Can you elaborate?

Reply:
When no error, the result is considered OK.

So you return it, so it's good.

But then, you assume implicitly that **any error** is about overflow.

But it could be something else.

I would have expected something like this

```suggestion
	q, r, err := tryQuoRemU128(d, e)
	if err == nil {
		return q, r, nil
	}

	if !errors.Is(err, errOverflow) {
		return Decimal{}, Decimal{}, err
	}
```

Or this 

```suggestion
	q, r, err := tryQuoRemU128(d, e)
	switch {
		case errors.Is(err, errOverflow):
			// OK, let's continue with bigint
		case err == nil {
			// we computed it
			return q, r, nil
		default:
			// anything else must be reported
			return Decimal{}, Decimal{}, err
	}
```

Reply:
It only returns `overflow` error so it's fine.

`errors.Is` is also much slower than comparing errors directly. I think it should be avoided to not affect the performance
```
goos: linux
goarch: amd64
pkg: github.com/quagmt/udecimal
cpu: Intel(R) Core(TM) i9-14900HX
BenchmarkErrorIs/errors.Is-32         	100420094	        11.79 ns/op	       0 B/op	       0 allocs/op
BenchmarkErrorIs/sentinel_error-32    	1000000000	         0.1929 ns/op	       0 B/op	       0 allocs/op
PASS
ok  	github.com/quagmt/udecimal	2.052s
```

Reply:
If only errOverflow is returned everywhere, maybe you could remove the error logic and return bool that will be true if it overflows.

Because it's not an error then, but a flag

Reply:
Boolean makes more sense in this case tbh, however using bool as a return value needs some more annotations/comments or named return values (I'm not a big fan) to explain it. I found `errOverflow` is self-explanatory and easier to understand. Do you have any suggestion?

Reply:
I agree with you, what about this

```suggestion
	q, r, err := tryQuoRemU128(d, e)
	switch {
		case err == nil {
			// we computed it
			return q, r, nil

		case err == errOverflow: // the only expected error
			// OK, let's continue with bigint

		default:
			// anything else must be reported
			return Decimal{}, Decimal{}, err
	}
```